File: app-java/src/main/java/com/java/bilibili/ui/main/MainActivity.java
package com.java.bilibili.ui.main;

import android.content.DialogInterface;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.widget.DrawerLayout;
import android.view.KeyEvent;
import android.view.MenuItem;
import android.widget.Toast;

import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.java.bilibili.R;
import com.java.bilibili.base.BaseActivity;
import com.java.bilibili.manager.AppManager;
import com.java.bilibili.ui.main.fragment.MainContentFragment;

import butterknife.BindView;

/**
 * Created by Eric.zhang on 2018/1/10.
 */

public class MainActivity extends BaseActivity implements BottomNavigationView.OnNavigationItemSelectedListener {
    @BindView(R.id.drawer_layout)
    DrawerLayout drawer_layout;
    @BindView(R.id.drawer)
    NavigationView drawer;

    @Override
    protected int getLayout() {
        return R.layout.ui_activity_main;
    }

    @Override
    protected void initEventAndData() {
//        drawer.setNavigationItemSelectedListener(this);
        switchFragment(0);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        return false;
    }

    private void switchFragment(int index) {
        Fragment fragment = null;
        String tag = null;
//        isShowHome = index == 0;
        switch (index) {
            case 0:
                tag = MainContentFragment.TAG;
                fragment = getSupportFragmentManager().findFragmentByTag(tag);
                if (fragment == null) {
                    fragment = MainContentFragment.newInstance();
                }
                break;
            case 1:
//                tag = PicContentFragment.TAG;
//                fragment = getSupportFragmentManager().findFragmentByTag(tag);
//                if (fragment == null) {
//                    fragment = PicContentFragment.newInstance();
//                }
                break;
            default:
                return;
        }
        getSupportFragmentManager().beginTransaction().replace(R.id.main_right_fragment, fragment, tag).commit();
    }

//    @Override
//    public boolean onKeyDown(int keyCode, KeyEvent event) {
//        if (drawer_layout.isDrawerOpen(drawer)) {
//            drawer_layout.closeDrawers();
//            return false;
//        }
//        switch (keyCode) {
//            case KeyEvent.KEYCODE_BACK:
//                new MaterialDialog.Builder(this)
//                        .title("退出")
//                        .content("是否退出app")
//                        .positiveText("是")
//                        .negativeText("否").onAny(new MaterialDialog.SingleButtonCallback() {
//                    @Override
//                    public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
//                        switch (which) {
//                            case POSITIVE:
//                                AppManager.getAppManager().AppExit(mActivity);
//                                break;
//                            case NEGATIVE:
//                                break;
//                        }
//                    }
//                })
//                        .show();
//                return false;
//        }
//        return super.onKeyDown(keyCode, event);
//    }
}

File: app-java/src/main/java/com/java/bilibili/ui/main/MainActivity.kt
package com.kotlin.bilibili.ui.main

import android.support.design.widget.NavigationView
import android.support.v4.app.Fragment
import android.view.KeyEvent
import android.view.MenuItem
import com.kotlin.bilibili.R
import com.kotlin.bilibili.R.id.*
import com.kotlin.bilibili.base.BaseActivity
import com.kotlin.bilibili.ui.main.fragment.MainContentFragment
import com.kotlin.bilibili.utils.Utils
import kotlinx.android.synthetic.main.ui_activity_main.*

class MainActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedListener {
    override fun layout(): Int = R.layout.ui_activity_main

    override fun initData() {
        drawer.setNavigationItemSelectedListener(this)
        switchFragment(0)
    }

    private fun switchFragment(index: Int) {
        var fragment: Fragment? = null
        var tag: String? = null
//        isShowHome = index == 0
        when (index) {
            0 -> {
                tag = "MainContentFragment"
                fragment = supportFragmentManager.findFragmentByTag(tag)
                if (fragment == null) {
                    fragment = MainContentFragment()
                }
            }
            1 -> {
//                tag = PicContentFragment.TAG
//                fragment = supportFragmentManager.findFragmentByTag(tag)
//                if (fragment == null) {
//                    fragment = PicContentFragment.newInstance()
//                }
            }
            else -> return
        }
        supportFragmentManager.beginTransaction().replace(R.id.main_right_fragment, fragment, tag).commit()
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            nav_home -> {

            }
            nav_histories -> {

            }
            nav_offline_manager -> {

            }
            nav_favorites -> {

            }
            nav_following -> {

            }
            nav_watch_later -> {

            }
            upper_center -> {

            }
            nav_live_center -> {

            }
            nav_vip -> {

            }
            nav_unicom_service -> {

            }
        }
        return true
    }

    var mkeyTime: Long = 0
    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (drawer_layout.isDrawerOpen(drawer)) {
            drawer_layout.closeDrawers()
            return false
        }
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            if (System.currentTimeMillis() - mkeyTime > 1200) {
                mkeyTime = System.currentTimeMillis()
            } else {
                Utils.exitApp(this)
            }
            return false
        }
        return super.onKeyDown(keyCode, event)
    }

    fun openDrawer() {
        drawer_layout.openDrawer(drawer)
    }
}
